state/prunning_service.go

```diff
@@ -0,0 +1,228 @@
+package state
```

Comment:
This file should probably be called `state/pruner.go` to coincide with the name of the primary struct provided.